src/components/image_upload.tsx

```diff
@@ -104,12 +102,20 @@ export class ImageUpload extends React.Component<ImageUploadProps, ImageUploadSt
 
         if (onFileRead) {
             const reader = new FileReader();
-            reader.onload = () => onFileRead(reader.result as string);
+            reader.onload = () => {
+                if (allowCropping) {
+                    this.setState({ cropURL: reader.result as string });
+                }
+                if (onFileRead) {
+                    onFileRead(reader.result as string);
+                }
+            };
             reader.onerror = console.error;
             reader.readAsDataURL(file);
         }
 
-        if (allowCropping) {
+        // This seems only to be used in test cases, as file.preview is not available in the browser
```

Comment:
That seems weird, I think we use preview for showing the preview while cropping?

Reply:
Nope! Preview is there, with this solution as well, as far as I can see.
Also makes no sense to crop the preview image. You want to crop the original image?

Reply:
This is why i think the other solution is better, but this keeps everything compatible.